.github/workflows/slurm_job_scripts/oci_cluster_manager.py

```diff
@@ -0,0 +1,165 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+import time
+import logging
+import sys
+import requests
+
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+
+_API_URL = os.environ['WEB_API_URL']
+_API_AUTH = requests.auth.HTTPBasicAuth('login', os.environ['WEB_API_TOKEN'])
+
+_SLEEP_TIME_SECONDS = int(os.environ.get("SLEEP_TIME_SECONDS", default="30"))
+_RETRY_PER_REGION = int(os.environ.get("RETRY_PER_REGION", default="3"))
+_REQUEST_TIMEOUT_SECONDS = int(os.environ.get("REQUEST_TIMEOUT_SECONDS", default=120))
+
+def get_regions():
+    return requests.post(_API_URL, json={'name':'list_regions'},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['regions']
+
+def find_existing_cluster():
+    return requests.post(_API_URL, json={'name':'find_cluster'},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['region']
+
+def get_cluster_ip(region):
+    return requests.post(_API_URL, json={'name':'get_cluster_ip', 'region':region},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['cluster_ip']
+
+def get_cluster_username(region):
+    return requests.post(_API_URL, json={'name':'get_cluster_username', 'region':region},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['cluster_username']
+
+def create_cluster(region):
+    logging.debug(requests.post(_API_URL, json={'name':'create_cluster', 'region':region},
+                                auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS))
+
+def add_pubkey(region, pubkey):
+    logging.debug(requests.post(_API_URL, json={'name':'add_pubkey', 'region':region, 'pubkey':pubkey},
+                                auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS))
+
+def get_cluster_hostkeys(region):
+    return requests.post(_API_URL, json={'name':'get_cluster_hostkeys', 'region':region},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['cluster_hostkeys']
+
+def get_status(region):
+    return requests.post(_API_URL, json={'name':'get_status', 'region': region},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['status']
+
+def destroy_all_clusters():
+    logging.debug(requests.post(_API_URL, json={'name':'destroy_all_clusters'},
+                                auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS))
+
+# This function is used to create and destroy clusters on demand.
```

Comment:
nit: put this big comment at the top of the file (below copyright) as a docstring. Also maybe give some usage instructions/examples?